### ISSUE TITLE
Moor build_runner breaks on commit c15c4738.

### ISSUE THREAD

**Issue by North101** [simolus3/moor#301]
I'm not sure why but moor fails to build my sql from c15c4738a629cd6459ab21d2076fe6321255e37b onwards. b928e34a1186275c9986e8bcd1cdc2cbd7b89b4b builds fine but if I try any later commits than that it fails with the following message:

```
> flutter packages pub run build_runner watch
[INFO] Generating build script...
[INFO] Generating build script completed, took 345ms

[INFO] Setting up file watchers...
[INFO] Setting up file watchers completed, took 13ms

[INFO] Waiting for all file watchers to be ready...
[INFO] Waiting for all file watchers to be ready completed, took 202ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 86ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 769ms

[WARNING] moor_generator|lib/src/model/specified_db_classes.dart was not found in the asset graph, incremental builds will not work.
 This probably means you don't have your dependencies specified fully in your pubspec.yaml.
[WARNING] Invalidating asset graph due to build script update!
[INFO] Cleaning up outputs from previous builds....
[INFO] Cleaning up outputs from previous builds. completed, took 5ms

[INFO] Terminating. No further builds will be scheduled

[INFO] Builds finished. Safe to exit

[INFO] Generating build script...
[INFO] Generating build script completed, took 65ms

[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 7.2s

[SEVERE] Failed to snapshot build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.
[SEVERE] /usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/analyzer/runner/results.dart:39:7: Error: Type 'TableInducingStatement' not found.  Map<TableInducingStatement, MoorTable> tableDeclarations;      ^^^^^^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/model/declarations/tables.dart:27:9: Error: Type 'TableInducingStatement' not found.  final TableInducingStatement node;        ^^^^^^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/model/declarations/tables.dart:31:32: Error: Type 'TableInducingStatement' not found.  factory MoorTableDeclaration(TableInducingStatement node, FoundFile file) {                               ^^^^^^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/analyzer/moor/create_table_reader.dart:13:9: Error: Type 'TableInducingStatement' not found.  final TableInducingStatement stmt;        ^^^^^^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/analyzer/runner/results.dart:39:7: Error: 'TableInducingStatement' isn't a type.  Map<TableInducingStatement, MoorTable> tableDeclarations;      ^^^^^^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/analyzer/session.dart:40:7: Error: No named parameter with the name 'enableJson1Module'.      enableJson1Module: options.hasModule(SqlModule.json1),      ^^^^^^^^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/hosted/pub.dartlang.org/sqlparser-0.4.0/lib/src/engine/sql_engine.dart:21:3: Context: Found this candidate, but the arguments don't match.  SqlEngine({this.useMoorExtensions = false}) {  ^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/model/sql_query.dart:44:31: Error: 'SimpleName' isn't a type.    assert(stmt.identifier is SimpleName);                              ^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/model/sql_query.dart:45:38: Error: 'SimpleName' isn't a type.    final name = (stmt.identifier as SimpleName).name;                                     ^^^^^^^^^^/usr/local/Caskroom/flutter/latest/flutter/.pub-cache/git/moor-c15c4738a629cd6459ab21d2076fe6321255e37b/moor_generator/lib/src/writer/tables/table_writer.dart:303:40: Error: 'CreateVirtualTableStatement' isn't a type.      final stmt = declaration.node as CreateVirtualTableStatement;     
```

pubspec.yaml
```
dependencies:
  moor_flutter:
    git:
      url: git@github.com:simolus3/moor.git
      ref: develop
      path: moor_flutter

dev_dependencies:
  build_runner:
  moor_generator:
    git:
      url: git@github.com:simolus3/moor.git
      ref: develop
      path: moor_generator

dependency_overrides:
  moor:
    git:
      url: git@github.com:simolus3/moor.git
      ref: develop
      path: moor
```

**Comment by simolus3** [comment]
If you're using a development version of moor_generator I think you would also use a development version of sqlparser. Try adding this to your `dependency_overrides`:
```yaml
  sqlparser:
    git:
      url: git@github.com:simolus3/moor.git
      ref: develop
      path: sqlparser
```
Then run `flutter packages get`, `flutter packages pub run build_runner clean` (just in case) and `flutter packages pub run build_runner build`.

**Comment by North101** [comment]
That worked. Thanks!

**Comment by North101** [comment]
@simolus3 This also fixed #287 for me. It's now generating `table.column` values.
